### ISSUE TITLE
Using aliases to edit, not overwrite, names

### ISSUE THREAD

**Issue by xathon** [roosta/i3wsr#30]
I have a program with a pretty long name, that I often have running in multiple instances. I'd like to be able to alias this in a way that, for example, `Window Title - Superlong Programname` becomes `Window Title - SLP`. Currently I can only find the option to have it appear as `SLP`, without catching the rest of the name of the window.

**Comment by roosta** [comment]
I'm not sure I understand the problem. Couldn't you just setup an alias for the whole thing `window title - Super long Programname`

If you mean that i3wsr should preserve parts of the title, how do you suppose that would work?

**Comment by xathon** [comment]
Yeah, sorry for not being clear. The `Window Title` part is dynamic, so it could be `foobar - Superlong Programname`, `qwerty - Superlong Programname` etc., and I'd like to catch only the `Superlong Programname` part. 
The aliases, if I understand correctly, currently work like this - if one alias matches the class/instance/name of the window, it replaces the whole name with the value entered as alias. From how the readme looks like, I'd expect to have quoted strings 
I understand that this might be a breaking change, but would it be possible to have at least the option to only have it replace the part of the property that the regex string matches? Like this:
```toml
# If the window title is 'New Tab - Google-chrome-unstable'
"^Google-chrome-unstable$" = "Chrome-dev" # doesn't match
"Google-chrome-unstable$" = "Chrome-dev" # new title is New Tab - Chrome-dev
".*Google-chrome-unstable" = "Chrome-dev" # new title is Chrome-dev
```

**Comment by roosta** [comment]
OK, I understand better now. I can create an option to enable replacing only the matched part of the regex. I'll address this and some other stuff when I get the chance. I'm a bit bogged down with projects ATM, but I plan on doing some serious work on i3wsr when things slow down a bit.

**Comment by spsjvc** [comment]
Hey @roosta, I was wondering about a similar use case.

I usually have multiple Visual Studio Code windows open for different projects across multiple workspaces. I'd like to "extract" the title of the project from the regex and just show an icon beside it, to better distinguish between what's open.

I'd be happy to help out with this with some guidance 🙂

**Comment by roosta** [comment]
Hi @spsjvc, I've been neglecting this issue for a while, thanks for showing an interest. If you want to put together a PR I'll gladly look it over, and get it merged. I don't really use a lot of title matching myself so the motivation to work on this has been a bit lacking, but it does present some interesting possibilities.

**Comment by rgallaispou** [comment]
Looks like this is revolved. Can it be closed ?

**Comment by roosta** [comment]
You're right, I must've forgot to close, thanks for pointing it out :)
